File: tests/fixtures/scan-codebase/sample.js
// sample.js - Test fixture for scan-codebase.cjs JS extraction
// Known imports and exports for deterministic testing

import express, { Router, json } from 'express';
import { hashPassword } from './utils/hash';
import config from '../config';

const path = require('path');
const fs = require('fs');

// Dynamic import
const lazyModule = import('./lazy-module');

// This comment has a fake import that should NOT be extracted:
// import fakeModule from 'should-not-appear';

// URL with :// that should NOT be stripped as a comment
const API_URL = 'https://api.example.com/v1';

export const API_VERSION = '2.0';

export function createServer(port) {
  return express().listen(port);
}

export class AppRouter {
  constructor() {
    this.router = Router();
  }
}

File: tests/fixtures/scan-codebase/sample.py
# sample.py - Test fixture for scan-codebase.cjs Python extraction
# Known imports and exports for deterministic testing

import os
import sys
from pathlib import Path
from typing import Optional, List
from .models import User
from ..utils import hash_password

# This is a commented import that should NOT be extracted:
# import fake_module

def create_user(name, email):
    """Create a new user."""
    return User(name=name, email=email)

class UserRepository:
    """Repository for user data."""
    def __init__(self):
        self.users = []
